### PR TITLE
Add tests against an actual trace agent.

### PR DESCRIPTION
**Pull request by tylerbenson**
@palazzem what other test cases do you think this should cover?

### CODE REVIEW

File: dd-trace-ot/src/traceAgentTest/groovy/DDApiIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      traces         | test
+      [""]           | 1
+      ["", 123]      | 2
+      [[:]]          | 3
+      [new Object()] | 4
+    }
+  }
+
+  static class DDApiIntegrationV3Test extends DDApiIntegrationV4Test {
+    boolean v4() {
```
Comment (realark) [inline dd-trace-ot/src/traceAgentTest/groovy/DDApiIntegrationTest.groovy]: :+1:

**Comment by realark** [comment]
@tylerbenson That's weird, I'm not sure about that one.


https://circleci.com/gh/DataDog/dd-trace-java/2906?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
```
java.io.IOException: Cannot run program "git" (in directory "/root/dd-trace-java/dd-java-agent"): error=2, No such file or directory
```

Looks like the directory was removed? That would explain why the artifacts couldn't be saved.

**Comment by tylerbenson** [comment]
That’s an unrelated problem to my question. I was asking about saving artifacts for the new job I created. I’ll look into what’s going on with this java 7 build.

**Comment by realark** [comment]
I see. `save_artifacts.sh` expects the results to be in `build/test-results`. Because the `traceAgentTest` task isn't run under the `test` task, is it possible the reports are ending up in another dir?

https://github.com/DataDog/dd-trace-java/blob/master/.circleci/save_artifacts.sh#L19

**Pull request merged**